precompiles/call-permit/src/lib.rs

```diff
@@ -243,12 +242,13 @@ where
 
 		let (reason, output) =
 			handle.call(to, transfer, data, Some(gas_limit), false, &sub_context);
-
 		match reason {
 			ExitReason::Error(exit_status) => Err(PrecompileFailure::Error { exit_status }),
 			ExitReason::Fatal(exit_status) => Err(PrecompileFailure::Fatal { exit_status }),
-			ExitReason::Revert(_) => Err(revert(output)),
-			ExitReason::Succeed(_) => Ok(succeed(output)),
+			ExitReason::Revert(_) => Err(revert(EvmDataWriter::new().write(Bytes(output)).build())),
```

Comment:
In case of revert the revert message must be returned as-is, neither encode (like here) or before (because due to a recent change the `revert` function does encoding too.
```suggestion
			ExitReason::Revert(_) => Err(PrecompileFailure::Revert {
				exit_status: ExitRevert::Reverted,
				output,
			}),
```